model/model_training/custom_datasets/formatting.py

```diff
@@ -15,20 +28,291 @@ def format_system_prefix(prefix, eos_token):
     )
 
 
-def format_pairs(pairs, eos_token, add_initial_reply_token=False):
-    conversations = [
-        "{}{}{}".format(QA_SPECIAL_TOKENS["Question" if i % 2 == 0 else "Answer"], pairs[i], eos_token)
-        for i in range(len(pairs))
-    ]
-    if add_initial_reply_token:
-        conversations.append(QA_SPECIAL_TOKENS["Answer"])
-    return conversations
+@unique
+class Language(str, Enum):
```

Comment:
worth considering extracting this class into a separate file. not necessary, i'll leave it up to you

Reply:
good point. Done